meta-client/src/main/java/org/apache/helix/metaclient/api/SubEntryChangeListener.java

```diff
@@ -0,0 +1,40 @@
+package org.apache.helix.metaclient.api;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/**
+ * Listener interface for children change events on a particular key. It includes new subentry
+ * creation, subentry deletion, subentry data change and when the listener is removed.
+ * This listener type can only be registered as a persist listener.
+ * For hierarchy key spaces like zookeeper, it refers to an entry's entire subtree.
+ * For flat key spaces, it refers to keys that matches `prefix*`.
+ */
+public interface SubEntryChangeListener {
```

Comment:
Same as commented before. I would suggest to use children instead of subentry. Because if this is not hierarchy data structure, then there is no sub entry concept. But if it is hierarchy structure, then children makes more sense as the world.

Even for prefix match, subentry is confusing word as "sub"

Reply:
For non hierarchy data structure, doesn't subentry means prefix matched entries?